FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +43,53 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] httpClientResult in
+
+			guard self != nil else { return }
+
+			var result: FeedLoader.Result
+
+			let feedLoaderResult = Self.httpClientResult2RemoteFeedLoaderResult(httpClientResult)
+			switch feedLoaderResult {
+			case .success(let responseData):
+				result = Self.responseData2FeedLoaderResult(responseData)
+
+			case .failure(let error):
+				result = .failure(error)
+			}
+
+			completion(result)
```

Comment:
Here, all the `switch` statement is doing is changing the success case from one type to another, while passing the same error. We can actually do that with `flatMap`:
```suggestion
			completion(Self.dataResult(from: httpClientResult)
				.flatMap(RemoteFeedLoader.feedLoaderResult))
```
(Also, note how `Self.dataResult(from: httpClientResult)` improves readability over `Self.httpClientResult2RemoteFeedLoaderResult(httpClientResult)`)